sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/InstrumentationRegistryHelper.java

```diff
@@ -6,22 +6,40 @@
 package com.microsoft.appcenter.utils;
 
 import android.os.Bundle;
-import android.support.test.InstrumentationRegistry;
+
+import java.lang.reflect.Method;
 
 /**
  * Wraps InstrumentationRegistry class to enable mocking in unit tests.
  */
 public class InstrumentationRegistryHelper {
 
+    private static final String[] LOCATIONS = new String[]{"androidx.test.platform.app.InstrumentationRegistry",
+            "androidx.test.InstrumentationRegistry",
+            "android.support.test.InstrumentationRegistry"};
+
     /**
      * Get the instrumentation arguments from the InstrumentationRegistry. Wrapper exists for unit
      * tests.
      *
      * @return the instrumentation arguments.
-     * @throws LinkageError          if the class, method is not found or does not match, typically no test dependencies in release.
-     * @throws IllegalStateException if no argument Bundle has been registered.
+     * @throws IllegalStateException if it cannot call getArguments using any of the provided classes.
      */
-    public static Bundle getArguments() throws LinkageError, IllegalStateException {
-        return InstrumentationRegistry.getArguments();
+    public static Bundle getArguments() throws IllegalStateException {
+        Exception exception = null;
+        for (String location : LOCATIONS) {
+            try {
+                Class<?> aClass = getClass(location);
+                Method getArguments = aClass.getMethod("getArguments");
+                return (Bundle) getArguments.invoke(null);
+            } catch (Exception e) {
+                exception = e;
+            }
+        }
+        throw new IllegalStateException(exception);
+    }
+
+    static Class<?> getClass(String className) throws ClassNotFoundException {
```

Comment:
Now you don't use it in test anymore, has to be inlined to make the warning go away.

Reply:
We do use this is in the test don't we?

```
        Method getClass = method(InstrumentationRegistryHelper.class, "getClass", String.class);
        doThrow(new ClassNotFoundException()).when(InstrumentationRegistryHelper.class, getClass)
```

Reply:
Why calling via reflection? Please increase visibility and add `@VisibleForTesting`.

Reply:
I struggled to get a mockito `mock` or `spy` on a static class which stubs `getClass` but not `getArguments` - I had another go at this but couldn't find a better way to do it using the API directly.

Reply:
I guess it's fine as is. There are other ways to do that but it's not worth it (isolate a utility and mockStatic, or mock the LOCATIONS with different values).